CMakeLists.txt

```diff
@@ -213,6 +213,13 @@ endif()
 # Apply the final optimization flag(s)
 zephyr_compile_options(${OPTIMIZATION_FLAG})
 
+if(CONFIG_LTO)
+  add_compile_options( -flto )
+  add_link_options( -flto )
```

Comment:
different toolchains uses different flags, thus this should be defined as properties here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/compiler_flags_template.cmake
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker/linker_flags_template.cmake

as the LTO indirectly has a dependency to gcc, because of this:
https://github.com/zephyrproject-rtos/zephyr/blob/7939cb83dc8ed8700b3ee6ca95fba08d420a977c/arch/Kconfig#L402

then it will be sufficient for now to only define the properties in gcc and ld.

For exact same reason, this adjustment is accepted as followup PR.

Reply:
Done